compose/config/environment.py

```diff
@@ -10,27 +10,32 @@
 
 from ..const import IS_WINDOWS_PLATFORM
 from .errors import ConfigurationError
+from .errors import EnvFileNotFound
 
 log = logging.getLogger(__name__)
 
 
 def split_env(env):
     if isinstance(env, six.binary_type):
         env = env.decode('utf-8', 'replace')
+    key, value = None, None
     if '=' in env:
-        return env.split('=', 1)
+        key, value = env.split('=', 1)
     else:
-        return env, None
+        key = env
+    if ' ' in key:
+        raise ConfigurationError("environment variable '%s' should not contains white spaces." % key)
+    return key, value
 
 
 def env_vars_from_file(filename):
     """
     Read in a line delimited file of environment variables.
     """
     if not os.path.exists(filename):
-        raise ConfigurationError("Couldn't find env file: %s" % filename)
+        raise EnvFileNotFound("Couldn't find env file: %s" % filename)
     elif not os.path.isfile(filename):
-        raise ConfigurationError("%s is not a file." % (filename))
+        raise EnvFileNotFound("%s is not a file." % (filename))
     env = {}
     with contextlib.closing(codecs.open(filename, 'r', 'utf-8-sig')) as fileobj:
         for line in fileobj:
```

Comment:
We should catch the `ConfigurationError` to provide more information about the file where the error occurred, so people can find the source of the issue more easily if they have several environment files.

```python
for line in fileobj:
  line = line.strip()
  if line and not line.startswith('#'):
    try:
      k, v = split_env(line)
      env[k] = v
    except ConfigurationError as e:
      raise ConfigurationError('In file {}: {}'.format(filename, e.msg)
```

Reply:
done.